containers/grpc-proxy/Dockerfile

```diff
@@ -1,2 +1,2 @@
-FROM envoyproxy/envoy-dev:933e67a9ecac814839f514fb5c31c78f3d33e5fe
```

Comment:
The old docker image (`933e67a`) was pushed a year ago, whereas the newer docker image (`cc089aa`) was pushed 18 hours ago and is the image listed in their [docker-compose documentation](https://www.envoyproxy.io/docs/envoy/latest/start/docker) (they must have a documentation and image build system thats coordinated, which is pretty cool. 

It's probably better to go with the newer docker image because it's probably a later version of Envoy (though I couldn't tell you which version just by looking at the Dockerfile) - later versions tend to have security updates. Unfortunately, the old configuration didn't work with the newer image, hence the change to the `envoy.yaml` configuration below. This works on docker compose, but it is untested on k8s; so when we deploy we need to check to make sure the proxy is working and if it doesn't we can roll back to the old config/image.